src/sentry/rules/conditions/event_frequency.py

```diff
@@ -125,3 +127,76 @@ def query_hook(self, event, start, end, environment_id):
             environment_id=environment_id,
             use_cache=True,
         )[event.group_id]
+
+
+percent_intervals = {
+    "1m": ("one minute", timedelta(minutes=1)),
+    "5m": ("five minutes", timedelta(minutes=5)),
+    "10m": ("ten minutes", timedelta(minutes=10)),
+    "30m": ("30 minutes", timedelta(minutes=30)),
+    "1h": ("one hour", timedelta(minutes=60)),
+}
+
+
+class EventFrequencyPercentForm(EventFrequencyForm):
+    intervals = percent_intervals
+    interval = forms.ChoiceField(
+        choices=[
+            (key, label)
+            for key, (label, duration) in sorted(
+                percent_intervals.items(),
+                key=lambda key____label__duration: key____label__duration[1][1],
+            )
+        ]
+    )
+    value = forms.IntegerField(widget=forms.TextInput())
+
+
+class EventFrequencyPercentCondition(BaseEventFrequencyCondition):
+    label = "The issue has more errors than {value} percent of sessions in {interval}"
+
+    def __init__(self, *args, **kwargs):
+        self.intervals = percent_intervals
+        self.form_cls = EventFrequencyPercentForm
+        super().__init__(*args, **kwargs)
+
+    def query_hook(self, event, start, end, environment_id):
+        project_id = event.project_id
+        cache_key = f"r.c.spc:{project_id}-{environment_id}-{start.replace(minute=0, second=0, microsecond=0, tzinfo=None)}-{end.replace(minute=0, second=0, microsecond=0, tzinfo=None)}".replace(
+            " ", ""
+        )
+        session_count = cache.get(cache_key)
+        if session_count is None:
+            filters = {"project_id": [project_id]}
+            if environment_id:
+                filters["environment"] = [environment_id]
+            result_totals = raw_query(
+                selected_columns=["sessions"],
+                rollup=60,
+                dataset=Dataset.Sessions,
+                start=start,
+                end=end,
+                filter_keys=filters,
+                groupby=["bucketed_started"],
+                referrer="rules.conditions.event_frequency.EventFrequencyPercentCondition",
+            )
+            if result_totals["data"]:
+                session_count = 0
+                for bucket in result_totals["data"]:
+                    session_count += bucket["sessions"]
+            else:
+                session_count = False
+            cache.set(cache_key, session_count, 600)
+
+        if session_count:
```

Comment:
If session_count is 0, but issue_count > 1, isn't that 100%? Or do you default to 0 in the case of the user not having the correct data?

Reply:
I think we're going to default to 0 here so that the alert doesn't fire in this case, at least for now.